src/pages/SignupPage.tsx

```diff
@@ -1,123 +1,126 @@
 import { useForm, SubmitHandler } from 'react-hook-form';
-import ErrorText from '@components/ErrorText';
-import Loader from '@components/Loader';
-import useSignUp from '@hooks/useSignUp';
+import { useMutation } from '@tanstack/react-query';
+import { axiosClient } from '@/api/axiosClient';
 
-type SignUpFormValues = {
+type FormValues = {
   email: string;
   password: string;
+  passwordConfirm: string;
   nickname: string;
-  passwordCheck: string;
 };
 
-const SignUpPage = () => {
+const SIGNUP_URL = '/signup';
+
+// 공식 문서 예제의 스타일을 가져와 사용
+const inputDefaultStyle = `mt-1 block w-full px-3 py-2 bg-white border border-slate-300 rounded-md text-sm shadow-sm placeholder-slate-400
+focus:outline-none focus:border-sky-500 focus:ring-1 focus:ring-sky-500
+disabled:bg-slate-50 disabled:text-slate-500 disabled:border-slate-200 disabled:shadow-none
+invalid:border-pink-500 invalid:text-pink-600
+focus:invalid:border-pink-500 focus:invalid:ring-pink-500`;
+const validationErrorStyle = `text-right text-sm font-bold text-red-700`;
+const buttonStyle = `rounded-lg bg-blue-500 px-4 py-2 text-white mt-5 w-[100%] hover:bg-blue-600`;
+const spanStyle = `mt-5 block text-left text-md font-bold text-slate-700`;
+
+const SignupPage = () => {
   const {
     register,
     handleSubmit,
     formState: { errors },
-  } = useForm<SignUpFormValues>();
-  const { signUpMutate, isLoading } = useSignUp();
+  } = useForm<FormValues>({
+    mode: 'onChange',
+  });
+
+  const userSignup = async (data: FormValues) => {
+    const response = await axiosClient.post(SIGNUP_URL, {
+      email: data.email,
+      password: data.password,
+      fullName: data.nickname,
+    });
```

Comment:
p3: 요쪽은 구조분해 할당 쓰는게 좋을 것 같아요 :)
```suggestion
  const userSignup = async ({email,password,nickname}: FormValues) => {
    const response = await axiosClient.post(SIGNUP_URL, {
      email,
      password,
      fullName: nickname,
    });
```